src/api/firebase.js

```diff
@@ -80,11 +82,38 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 
 export async function updateItem(listId, itemId) {
 	const docRef = doc(db, listId, itemId);
+	// We need to use some data from the document for our future purchase estimates.
+	const docSnap = await getDoc(docRef);
+	const today = new Date();
 
-	await updateDoc(docRef, {
-		dateLastPurchased: new Date(),
-		totalPurchases: increment(1),
-	});
+	if (docSnap.exists()) {
+		const data = docSnap.data();
+		// If the user has never purchased this item before, we'll use the date
+		// it was created as the date of the last purchase.
+		const daysBetweenPurchases = getDaysBetweenDates(
+			data.dateLastPurchased?.toDate() ?? data.dateCreated.toDate(),
+			today,
+		);
+		// If the user has never purchased this item before then we will pass
+		// in an undefined value and previousEstimate will return 14 days, the
+		// default used in calculateEstimate.
+		const previousEstimate = getDaysBetweenDates(
+			data.dateLastPurchased?.toDate(),
+			data.dateNextPurchased.toDate(),
+		);
+		const newEstimateDate = getFutureDate(
+			calculateEstimate(
+				previousEstimate,
+				daysBetweenPurchases,
+				data.totalPurchases,
```

Comment:
Take a look [https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils?activeTab=code](url)

Really minor, but shouldn't total purchases be current purchase + 1, calculateEstimate assumes totalPurchases includes this current purchase.